src/containers/dashboard/player.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import { PlayerType } from './types';
+
+type OnChangeType = { type: 'NAME', value: string } | { type: 'AGE', value: number };
+
+const Player = ({ player, onSaveUpdate, isInput = false }: { player: PlayerType, onSaveUpdate: (player: PlayerType) => void, isInput?: boolean }) => {
+	const [name, setName] = useState<string>(player.name || '');
+	const [age, setAge] = useState<number | null>(player.age);
+	const [enableSaveBtn, setEnableSaveBtn] = useState<boolean>(false);
+	const [error, setError] = useState<string>('');
+
+
+	const onChange = ({ type, value }: OnChangeType) => {
+	  if (type === 'AGE') {
+			setAge(value);
+	} else {
+			setName(value);
+	}
+	  setEnableSaveBtn(validateInputs(type, value));
+	};
+
+	const validateInputs = (type: string, value: any): boolean => {
+	  const res = (type === 'AGE') ? (value >= 15 && value <= 60) : /^[a-zA-Z0-9 ]{3,}$/.test(value);
+	  setError(res ? '' : 'The name must be alphanumeric, and the age should be between 15 and 60.');
+	  return res;
+	}
+
+
+	return (<div>
+		<div style={{ display: 'flex', justifyContent: 'space-between', gap: '5px', margin: '5px' }}>
+			<input
+				type='text'
+				placeholder='Player Name'
+				value={name}
+				style={{ width: '100px' }}
+				onChange={(e: React.ChangeEvent<HTMLInputElement>) => onChange({ type: 'NAME', value: e.target.value })}
+			/>
+			<input
+				type='number'
+				placeholder='age'
+				min={15} max={60}
+				value={age || ''}
+				style={{ width: '50px' }}
+				onChange={(e: React.ChangeEvent<HTMLInputElement>) => onChange({ type: 'AGE', value: Number(e.target.value) })}
+			/>
+			<button
+				disabled={!enableSaveBtn}
```

Comment:
@Pathy-Subhamraj can you please give a thought on this approach, do we need enableSaveBtn flag or validating the inputs on every render is better?

cc: @DipankarNath

Reply:
@cb-rohitsharma , Technically it is calling the validate function on every input key stroke. I can put a debounce to reduce the number of calls. But we will need the function to do the validation.

Reply:
Or, maybe I can do the call on button click. we stop the operation if there are errors, else process through. That will be efficient. But a different UX.